health/notifications/alarm-notify.sh.in

```diff
@@ -1664,6 +1675,55 @@ send_awssns() {
 	return 1
 }
 
+# -----------------------------------------------------------------------------
+# matrix sender
+
+send_matrix() {
+    local homeserver="${1}" webhook accesstoken rooms="${2}" httpcode sent=0 payload color
+
+	[ "${SEND_MATRIX}" != "YES" ] && return 1
+
+    # check if access token was provided
+    if [ "${SEND_MATRIX}" == "YES"  ] && [ -n "${MATRIX_ACCESSTOKEN}" ]; then
+        accesstoken="${MATRIX_ACCESSTOKEN}"
+    else
+        return 1
+    fi
+
+	case "${status}" in
+	WARNING) color="warning" ;;
+	CRITICAL) color="danger" ;;
+	CLEAR) color="good" ;;
+	*) color="#777777" ;;
+	esac
+
+	for room in ${rooms}; do
+        webhook="$homeserver/_matrix/client/r0/rooms/$(urlencode $room)/send/m.room.message?access_token=$accesstoken"
+		payload="$(
+        cat <<EOF
+        {
+            "msgtype": "m.notice",
+            "format": "org.matrix.custom.html",
+            "formatted_body": "<font color=\"$color\">${host} ${status_message}, \`${chart}\` (${family}), <b>${alarm}</b></font>",
+            "body": "${host} ${status_message}, \`${chart}\` (${family}), *${alarm}*"
+        }
+EOF
+		)"
+
+		httpcode=$(docurl -X POST --data-urlencode "payload=${payload}" "${webhook}")
```

Comment:
I finished testing.

I kept getting errors when I ran the curl command manually:

```
{
    "errcode": "M_NOT_JSON",
    "error": "Content not JSON."
}
```
I tried various things and the only way I could get it to work was to change the `--data-urlencode "payload=...` to `--data "..."` (no url encoding and no `payload=`).
The curl examples at https://matrix.org/docs/guides/client-server.html are a bit strange, but it looks like this is necessary.